lib/elixir_sense/core/state.ex

```diff
@@ -246,15 +243,27 @@ defmodule ElixirSense.Core.State do
 
   def add_import(state, module) do
     [imports_from_scope|inherited_imports] = state.imports
+    imports_from_scope = imports_from_scope -- [module]
     %{state | imports: [[module|imports_from_scope]|inherited_imports]}
+    |> maybe_add_import_alias(module)
   end
 
   def add_imports(state, modules) do
     Enum.reduce(modules, state, fn(mod, state) -> add_import(state, mod) end)
   end
 
+  defp maybe_add_import_alias(state, module) do
+    case module |> Atom.to_string() |> String.split(".") |> Enum.map(& &1 |> String.to_atom()) |> Enum.drop(1) |> Enum.reverse do
```

Comment:
Worth extracting this to a variable/function, I think.

Reply:
Agreed, i'll refactor this function